sssd_test_framework/utils/sbus.py

```diff
@@ -0,0 +1,476 @@
+        if iface is None:
+            raise RuntimeError(f"No interface found for {child.attrib['name']}")
+
+        input = self._get_method_args(child, "in")
+        output = self._get_method_args(child, "out")
+
+        method = ProxyMethod(
+            self.host,
+            self._dest,
+            self._objpath,
+            self._bus,
+            interface=iface.attrib["name"],
+            child=child,
+            input=input,
+            output=output,
+        )
+        self.__setattr__(child.attrib["name"], method)
+        return method
+
+    def _make_node(self, child) -> DBUSObject:
+        """
+        Add a D-Bus child node to the current node.
+        """
+        node = DBUSObject(
+            self.host, objpath=self._objpath + "/" + child.attrib["name"], dest=self._dest, bus=self._bus
+        )
+        self.__setattr__(child.attrib["name"], node)
+        return node
+
+    def _make_attr(self, child):
+        """
+        Make a D-Bus attribute and add it to the current node.
+        """
+        match child.tag:
+            case "property":
+                obj = self._make_prop(child)
+            case "method":
+                obj = self._make_method(child)
+            case "node":
+                obj = self._make_node(child)
+            case _:
+                raise AttributeError(
+                    f"'{self.__class__.__name__}' object has an attribute '{child.attrib['name']}'"
+                    f" of unkown type '{child.tag}'"
+                )
+
+        return obj
+
+    def __getattr__(self, attr) -> Any:
+        """
+        When acccessing an attribute that doesn't belong to the object,
+        create it if it should exist.
+        """
+
+        # This check avoids extra and recursive calls during object initialization
+        if "_properties" in self.__dict__ and attr in self._properties:
+            return self._properties[attr].get_value()
+
+        # This check avoids extra and recursive calls during object initialization
+        # if "_find_child" in self.__dict__ and "_xml" in self.__dict__:
+        if "_xml" in self.__dict__:
+            c = self._find_child_node(self._xml, attr)
+            if c is not None:
+                return self._make_attr(c)
+
+        raise AttributeError(f"'{self.__class__.__name__}' object has no attribute '{attr}'")
+
+    def __setattr__(self, attr: str, value: Any) -> None:
+        """
+        If the caller sets a property, invoke the set_value() method
+        of the corresponding proxy property.
+        """
+        # This check avoids extra and recursive calls during object initialization
+        if "_properties" in self.__dict__ and attr in self._properties:
+            return self._properties[attr].set_value(value)
+
+        return super().__setattr__(attr, value)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(*args):
+        pass
+
+
+class DBUSDestination:
```

Comment:
This should be MultihostUtility.

Reply:
Yep. Forgot to change that when I created this class.